### PR TITLE
Container tests

### PR DESCRIPTION
**Pull request by agniiyer**

### CODE REVIEW

File: ivy_tests/test_ivy/test_container.py [hunk 1]
```diff
@@ -532,54 +529,6 @@ def test_container_structural_diff(device):
     assert container_diff_same_only.to_dict() == container_diff.to_dict()
 
 
-def test_container_from_dict(device):
```
Comment (djl11) [inline ivy_tests/test_ivy/test_container.py:535]: I think we should keep this test, it basically tests the constructor `__init__` of the container when a `dict` is passed.

File: ivy/container/base.py [hunk 2]
```diff
@@ -3162,7 +3162,7 @@ def map_conts(
     def with_entries_as_lists(self):
         def to_list(x, _=""):
             try:
-                return self._ivy.to_list(x)
+                return x
```
Comment (djl11) [inline ivy/container/base.py:3165]: are you sure about this? The original seems correct to me.

File: ivy_tests/test_ivy/test_container.py [hunk 3]
```diff
@@ -532,54 +529,6 @@ def test_container_structural_diff(device):
...
-    container = Container(dict_in)
-    assert np.allclose(ivy.to_numpy(container["a"]), np.array([1]))
-    assert np.allclose(ivy.to_numpy(container.a), np.array([1]))
-    assert np.allclose(ivy.to_numpy(container["b"]["c"]), np.array([2]))
-    assert np.allclose(ivy.to_numpy(container.b.c), np.array([2]))
-    assert np.allclose(ivy.to_numpy(container["b"]["d"]), np.array([3]))
-    assert np.allclose(ivy.to_numpy(container.b.d), np.array([3]))
-
-
-def test_container_depth(device):
```
Comment (djl11) [inline ivy_tests/test_ivy/test_container.py:549]: I think we should keep this.

File: ivy_tests/test_ivy/test_container.py [hunk 4]
```diff
@@ -817,68 +766,6 @@ def test_container_show_sub_container(device):
     top_cont.show_sub_container(sub_cont)
 
 
-def test_container_from_dict_w_cont_types(device):
```
Comment (djl11) [inline ivy_tests/test_ivy/test_container.py:820]: I think we should keep this.

File: ivy_tests/test_ivy/test_container.py [hunk 5]
```diff
@@ -817,68 +766,6 @@ def test_container_show_sub_container(device):
...
-    container = Container(dict_in)
-    assert np.allclose(ivy.to_numpy(container["a"]), np.array([1]))
-    assert np.allclose(ivy.to_numpy(container.a), np.array([1]))
-    assert np.allclose(ivy.to_numpy(container["b"]["c"]), np.array([2]))
-    assert np.allclose(ivy.to_numpy(container.b.c), np.array([2]))
-    assert np.allclose(ivy.to_numpy(container["b"]["d"]), np.array([3]))
-    assert np.allclose(ivy.to_numpy(container.b.d), np.array([3]))
-
-
-def test_container_from_kwargs(device):
```
Comment (djl11) [inline ivy_tests/test_ivy/test_container.py:841]: I think we should keep this.

File: ivy_tests/test_ivy/test_container.py [hunk 6]
```diff
@@ -817,68 +766,6 @@ def test_container_show_sub_container(device):
...
-    )
-    assert np.allclose(ivy.to_numpy(container["a"]), np.array([1]))
-    assert np.allclose(ivy.to_numpy(container.a), np.array([1]))
-    assert np.allclose(ivy.to_numpy(container["b"]["c"]), np.array([2]))
-    assert np.allclose(ivy.to_numpy(container.b.c), np.array([2]))
-    assert np.allclose(ivy.to_numpy(container["b"]["d"]), np.array([3]))
-    assert np.allclose(ivy.to_numpy(container.b.d), np.array([3]))
-
-
-def test_container_from_list(device):
```
Comment (djl11) [inline ivy_tests/test_ivy/test_container.py:854]: I think we should keep this.

File: ivy_tests/test_ivy/test_container.py [hunk 7]
```diff
@@ -2304,31 +2189,6 @@ def test_container_reshape_like(device):
     assert list(container_reshaped.b.d.shape) == [3, 3, 1, 1]
 
 
-def test_container_slice(device):
```
Comment (djl11) [inline ivy_tests/test_ivy/test_container.py:2307]: I think we should keep this.

File: ivy_tests/test_ivy/test_container.py [hunk 8]
```diff
@@ -2472,35 +2332,6 @@ def test_container_to_disk_shuffle_and_from_disk_as_hdf5(device):
     os.remove(save_filepath)
 
 
-def test_container_pickle(device):
```
Comment (djl11) [inline ivy_tests/test_ivy/test_container.py:2475]: I think we should keep this.

**Review by djl11 — CHANGES_REQUESTED** [review]
I think most of the tests you removed should actually be kept. Many of these are testing special methods, operators, and the constructor etc. all of which we should keep. Perhaps the task description was unclear. Could you please explain your rationale for removing these tests :) thanks!

File: ivy_tests/test_ivy/test_container.py [hunk 9]
```diff
@@ -2550,1308 +2381,109 @@ def test_container_to_and_from_disk_as_json(device):
...
-    assert np.allclose(ivy.to_numpy(container.a), np.array([[[1.0], [2.0], [3.0]]]))
-    container.create_if_absent("e", ivy.array([[[4.0], [8.0], [12.0]]]), True)
-    assert np.allclose(ivy.to_numpy(container.e), np.array([[[4.0], [8.0], [12.0]]]))
-
-    # depth 2
-    container.create_if_absent("f/g", np.array([[[5.0], [10.0], [15.0]]]), True)
-    assert np.allclose(ivy.to_numpy(container.f.g), np.array([[[5.0], [10.0], [15.0]]]))
-
-
-def test_container_if_exists(device):
```
Comment (saeedashrraf) [inline ivy_tests/test_ivy/test_container.py:3699]: This should be kept

File: ivy_tests/test_ivy/test_container.py [hunk 10]
```diff
@@ -2550,1308 +2381,109 @@ def test_container_to_and_from_disk_as_json(device):
...
-        "b": {
-            "c": ivy.array([[[2.0], [4.0], [6.0]]], device=device),
-            "d": ivy.array([[[3.0], [6.0], [9.0]]], device=device),
-        },
-    }
-    container = Container(dict_in)
-    assert container.dev_str == device
-
-
-def test_container_create_if_absent(device):
```
Comment (saeedashrraf) [inline ivy_tests/test_ivy/test_container.py:3678]: This should be kept

File: ivy_tests/test_ivy/test_container.py [hunk 11]
```diff
@@ -2550,1308 +2381,109 @@ def test_container_to_and_from_disk_as_json(device):
...
-        ivy.to_numpy(container.if_exists("c")), np.array([[[1.0], [2.0], [3.0]]])
-    )
-    assert container.if_exists("d") is None
-    container.d = ivy.array([[[1.0], [2.0], [3.0]]], device=device)
-    assert np.allclose(
-        ivy.to_numpy(container.if_exists("d")), np.array([[[1.0], [2.0], [3.0]]])
-    )
-
-
-def test_jax_pytree_compatibility(device):
```
Comment (saeedashrraf) [inline ivy_tests/test_ivy/test_container.py:3724]: This should be kept

File: ivy_tests/test_ivy/test_container.py [hunk 12]
```diff
@@ -2550,1308 +2381,109 @@ def test_container_to_and_from_disk_as_json(device):
...
-
-    # dict flattened
-    true_values = tree_flatten(dict_in)[0]
-
-    # assertion
-    for i, true_val in enumerate(true_values):
-        assert np.array_equal(ivy.to_numpy(cont_values[i]), ivy.to_numpy(true_val))
-
-
-def test_container_from_queues(device):
```
Comment (saeedashrraf) [inline ivy_tests/test_ivy/test_container.py:3752]: This should be kept

**Review by saeedashrraf — CHANGES_REQUESTED** [review]

**Comment by agniiyer** [comment]
@djl11 Ah ok you mentioned on Discord that we should only test for the functions in base.py - nothing more, nothing less. So I removed the tests that didn't have corresponding names in base.py. I will keep all the functions mentioned here and request a review again :)

File: ivy/container/base.py (hunk 2 above)
Comment (agniiyer) [reply ivy/container/base.py:3165]: The original throws an error when the container has a string saying the string object can't be used with to_list(). The new version doesn't have this problem. Is there a better way to handle this?

**Pull request merged**